test/cql-pytest/test_allow_filtering.py

```diff
@@ -308,6 +308,42 @@ def table4(cql, test_keyspace):
 
 # Selecting from indexed table should using only clustering key should require filtering
 # Variation of #7608 found in #8991
-@pytest.mark.xfail(reason="Select from indexed table using only clustering key should require filtering. Issue #7608")
 def test_select_indexed_cluster(cql, table4):
     check_af_mandatory(cql, table4, 'c1 = 1 AND c2 = 1', lambda row: row.c1 == 1 and row.c2 == 1)
+
+# table5 contains an indexed table with 3 clustering columns.
+# used to test need_filtering() on indexed tables.
+@pytest.fixture(scope="module")
+def table5(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int, c1 text, c2 text, c3 int, PRIMARY KEY (p,c1,c2,c3))")
+    cql.execute(f"CREATE INDEX ON {table} (c3)")
+    cql.execute(f"INSERT INTO {table} (p, c1, c2, c3) VALUES (0, 'a', 'b', 0)")
+    cql.execute(f"INSERT INTO {table} (p, c1, c2, c3) VALUES (0, 'a', 'b', 1)")
+
+    everything = list(cql.execute(f"SELECT * FROM {table}"))
+    wait_for_index(cql, table, 'c3', everything)
+    yield (table, everything)
+    cql.execute(f"DROP TABLE {table}")
+
+# Test that new implementation of need_filtering() for indexes works correctly.
```

Comment:
In a year we won't remember what this "new" is. Please refer to some issue or pull request number.